File: python/queen-attack/queen_attack.py
class Queen(object):
    def __init__(self, row, column):
        if row < 0 or row > 7:
            raise ValueError('bad row value')
        if column < 0 or column > 7:
            raise ValueError('bad column value')
        self.row = row
        self.column = column

    def can_attack(self, another_queen):
        if self.row == another_queen.row and self.column == another_queen.column:
            raise ValueError('same position')

        # orthogonal
        if self.row == another_queen.row or self.column == another_queen.column:
            return True
        
        # diagonal
        if abs(self.row - another_queen.row) == abs(self.column - another_queen.column):
            return True
        
        return False

File: csharp/queen-attack/QueenAttack.cs
using System;

public class Queen
{
    public Queen(int row, int column)
    {
        Row = row;
        Column = column;
    }

    public int Row { get; }
    public int Column { get; }
}

public static class QueenAttack
{
    public static bool CanAttack(Queen white, Queen black)
    {
        return white.Row == black.Row || white.Column == black.Column ||
            Math.Abs(white.Row - black.Row) == Math.Abs(white.Column - black.Column);
    }

    public static Queen Create(int row, int column)
    {
        if (row < 0 || row > 7 || column < 0 || column > 7) {
            throw new ArgumentOutOfRangeException("bad stuff");
        }
        return new Queen(row, column);
    }
}